src/platform.ts

```diff
@@ -156,41 +172,58 @@ export class ElectroluxDevicesPlatform implements DynamicPlatformPlugin {
-                this.log.warn('Accessory not found for model: ', appliance.applianceData.modelName);
-                return;
-            }
+                const uuid = this.api.hap.uuid.generate(appliance.applianceId);
+
+                const existingAccessory = this.accessories.find(
+                    (accessory) => accessory.platformAccessory.UUID === uuid,
+                );
 
-            const uuid = this.api.hap.uuid.generate(appliance.applianceId);
+                if (existingAccessory) {
+                    this.log.info(
+                        'Restoring existing accessory from cache:',
+                        existingAccessory.platformAccessory.displayName,
+                    );
+                    existingAccessory.controller = new DEVICES[appliance.applianceData.modelName](
+                        this,
+                        existingAccessory.platformAccessory,
+                        appliance,
+                    );
+                    return;
+                }
 
-            const existingAccessory = this.accessories.find((accessory) => accessory.platformAccessory.UUID === uuid);
+                return {uuid, appliance};
+            })
+            .filter((entry): entry is {uuid: string; appliance: Appliance} => !!entry);
 
-            if (existingAccessory) {
-                this.log.info(
-                    'Restoring existing accessory from cache:',
-                    existingAccessory.platformAccessory.displayName,
-                );
-                existingAccessory.controller = new DEVICES[appliance.applianceData.modelName](
-                    this,
-                    existingAccessory.platformAccessory,
-                    appliance,
-                );
-                return;
-            }
+        if (newAppliances.length > 0) {
+            const applianceIds = newAppliances.map(({appliance}) => appliance.applianceId);
+            const appliancesInfo = await this.getAppliancesInfo(applianceIds);
 
-            this.log.info('Adding new accessory:', appliance.applianceData.applianceName);
+            for (const {uuid, appliance} of newAppliances) {
+                this.log.info('Adding new accessory:', appliance.applianceData.applianceName);
 
-            const platformAccessory = new this.api.platformAccessory(appliance.applianceData.applianceName, uuid);
-            const accessory = new ElectroluxAccessory(
-                platformAccessory,
-                new DEVICES[appliance.applianceData.modelName](this, platformAccessory, appliance),
-            );
-            this.accessories.push(accessory);
+                const info = appliancesInfo.find((info) => info.pnc === pnc(appliance.applianceId));
+                const platformAccessory = new this.api.platformAccessory<ElectroluxAccessoryContext>(
+                    appliance.applianceData.applianceName,
+                    uuid,
+                );
+                platformAccessory.context.info = info;
```

Comment:
Since we store the info in the accessory context, it will be missing for cached accessories.

I wanted to keep the logic simple, so I didn't bother adding cache invalidation.